File: 1574-maximum-product-of-two-elements-in-an-array/maximum-product-of-two-elements-in-an-array.go
func maxProduct(nums []int) int {
    max := 1
    max2 := 1;

    for _, v := range nums {
        if v >= max {
            max2 = max;
            max = v;
        } else if v >= max2 {
            max2 = v;
        }
    }
    
    return (max - 1) * (max2 - 1);
}

File: 1574-maximum-product-of-two-elements-in-an-array/maximum-product-of-two-elements-in-an-array.ts
function maxProduct(nums: number[]): number {
    let a = -Infinity, b = -Infinity;
    for(let i = 0; i < nums.length; i++) {
        if(nums[i] > a) {
            b = a;
            a = nums[i]
        } else if(nums[i] > b) b = nums[i]
    }
    return (a - 1) * (b - 1)
};